app/workers/job_concurrency.rb

```diff
@@ -36,7 +36,24 @@ module JobConcurrency
   ##
   # Run the concurrency check of good_job without actually trying to enqueue it
   # Will call the provided block in case the job would be cancelled
-  def check_concurrency(&block)
-    good_job_enqueue_concurrency_check(self, on_abort: block, on_enqueue: nil)
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/PerceivedComplexity
+  def check_concurrency
+    enqueue_limit = self.class.good_job_concurrency_config[:enqueue_limit]
+    enqueue_limit = instance_exec(&enqueue_limit) if enqueue_limit.respond_to?(:call)
+    enqueue_limit = nil unless enqueue_limit.present? && (0...Float::INFINITY).cover?(enqueue_limit)
+
+    unless enqueue_limit
+      total_limit = self.class.good_job_concurrency_config[:total_limit]
+      total_limit = instance_exec(&total_limit) if total_limit.respond_to?(:call)
+      total_limit = nil unless total_limit.present? && (0...Float::INFINITY).cover?(total_limit)
+    end
+
+    limit = enqueue_limit || total_limit
+    enqueued_jobs = GoodJob::Job.where(concurrency_key: good_job_concurrency_key).unfinished.advisory_unlocked.count
+
+    yield if limit.present? && enqueued_jobs + 1 > limit
```

Comment:
This does what it promises. I wonder though, if it could be written in a clearer way. Basically, the code repeats itself, so it could be changed to 

```
def check_concurrency
  limit = configured_concurrency_limit(self.class.good_job_concurrency_config[:enqueue_limit]) || configured_concurrency_limit(self.class.good_job_concurrency_config[:total_limit])
  enqueued_jobs = GoodJob::Job.where(concurrency_key: good_job_concurrency_key).unfinished.advisory_unlocked.count

  yield if limit.present? && enqueued_jobs + 1 > limit
end

def configured_concurrency_limit(limit)
  configured_limit = instance_exec(&limit) if limit.respond_to?(:call)
  configured_limit = nil unless limit.present? && (0...Float::INFINITY).cover?(limit)
  configured_limit
end

Reply:
:green_circle: Readability question from my side: I am confused about the `enqueued_jobs + 1 > limit`. Am I missing an obvious reason to not write this as `enqueued_jobs >= limit`?

Reply:
I've tried to change the code and didn't get the results I wanted. This was lifted directly from GoodJob internals, so I assume they have a reason for all of that.

That being said, what I want is to remove the dependency of this code ASAP rather than try to fix it.